xpk.py

```diff
@@ -3753,13 +3732,20 @@ def set_cluster_command(args) -> int:
   Returns:
     0 if successful and 1 otherwise.
   """
-  command = (
-      'gcloud container clusters get-credentials'
-      f' {args.cluster} --region={zone_to_region(args.zone)}'
-      f' --project={args.project} &&'
-      ' kubectl config view && kubectl config set-context --current'
-      ' --namespace=default'
-  )
+  if args.device_type == h150_device_type:
```

Comment:
@Obliviour Here we need to add `--device-type` back to workload deletion, because A3 and A3+ handle zone/region differently.

Reply:
`get-credentials` command is searching for the cluster. XPK only creates regional clusters so it should only need to search for a cluster using --region arg.

---

A scenario where you would need --zone is if the cluster was created outside of xpk using --zone. I am guessing that is what happened in the h150 cluster you are testing with. So we shouldn't need a different code path.